.github/workflows/build-deploy-microservices.yml

```diff
@@ -0,0 +1,279 @@
+          war_name: "Artemis.war"
+        - image_name: "artemis-gateway"
+          dockerfile_path: "gateway/src/main/docker"
+          war_name: "gateway.war"
+        - image_name: "artemis-usermanagement-service"
+          dockerfile_path: "user-management/src/main/docker"
+          war_name: "user-management.war"
+        - image_name: "artemis-lecture-service"
+          dockerfile_path: "lecture/src/main/docker"
+          war_name: "lecture.war"
+    steps:
+    - name: Download Artifact
+      uses: actions/download-artifact@v2
+      with:
+        name: ${{ matrix.war_name }}
+    - name: Download Dockerfile
+      run: curl -O ${{ env.RAW_URL }}/${{ matrix.dockerfile_path }}/Dockerfile -O ${{ env.RAW_URL }}/${{ matrix.dockerfile_path }}/entrypoint.sh
+    - name: Build Docker Image
+      run: docker build --build-arg WAR_PATH=. -t ${{ matrix.image_name }} .
+
+    - name: Compute Tag
+      uses: actions/github-script@v5
+      id: compute-tag
+      with:
+        result-encoding: string
+        script: |
+          if (context.eventName === "pull_request") {
+            return "pr-" + context.issue.number;
+          }
+          if (context.eventName === "push") {
+            if (context.ref.startsWith("refs/tags/")) {
+              return context.ref.slice(10);
+            }
+            if (context.ref === "refs/heads/develop") {
+              return "latest";
+            }
+          }
+          return "FALSE";
+
+    # Push to GitHub Container Registry
+    - name: Login to GitHub Container Registry
+      uses: docker/login-action@v1
+      with:
+        registry: ghcr.io
+        username: ${{ github.actor }}
+        password: ${{ secrets.GITHUB_TOKEN }}
+      if: ${{ steps.compute-tag.outputs.result != 'FALSE' }}
+    - name: Push to GitHub Container Registry
+      env:
+        TAG: ${{ steps.compute-tag.outputs.result }}
+      run: |
+        docker tag ${{ matrix.image_name }} ghcr.io/ls1intum/${{ matrix.image_name }}:$TAG
+        docker push ghcr.io/ls1intum/${{ matrix.image_name }}:$TAG
+      if: ${{ steps.compute-tag.outputs.result != 'FALSE' }}
+
+    # TODO: Push to Docker Hub (develop + tag)
+
+    # TODO: Push to Chair Habour (??)
+    #
+  deploy:
+    needs: docker_push_images
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        include:
+          - environment: artemistest6
+            url: https://artemistest6.ase.in.tum.de
+    environment:
+      url: ${{ matrix.url }}
+      name: ${{ matrix.environment }}
+
+    steps:
+    - uses: actions/checkout@v2
+
+    - name: Compute Tag
+      uses: actions/github-script@v5
+      id: compute-tag
+      with:
+        result-encoding: string
+        script: |
+          if (context.eventName === "pull_request") {
+            return "pr-" + context.issue.number;
+          }
+          if (context.eventName === "push") {
+            if (context.ref.startsWith("refs/tags/")) {
+              return context.ref.slice(10);
```

Comment:
Small suggestion: Instead of `10` it may be more explicit to use `"refs/tags/".length`, or assigning that string to a variable can be even better. On the other hand, this string looks highly unlikely to change nor this is introduced in the current PR so I guess this can be ignored.

Reply:
I think to leave it as it is in order to keep it consistent as it is used in the same way in other workflows.